### ISSUE TITLE
application.wadl resource path is absolute

### ISSUE THREAD

**Issue by vguna** [stoicflame/enunciate#220]
Generating the application.wadl with 2.1.1 leads to the problem, that the path attribute of the resource element is generated absolute. Example:

```xml
...
<wadl:resources base="http://localhost:8080/myapp/">
	<wadl:resource path="/api/1/accounts">
...
```
Per wadl spec (http://www.w3.org/Submission/wadl/#x3-120002.6) this should be a relative path.

That leads also to the problem, when clicking on the endpoint within the generated documentation (performing the request), that the context is dropped. So the browser invokes `http://localhost:8080/api/1/accounts` instead of `http://localhost:8080/myapp/api/1/accounts`. 

I've worked around (?) that by specifying the context explicitly on the jaxrs config again:

```xml
...
<jaxrs>
	<application path="/myapp/api/1" />
</jaxrs>.
...
```

I wonder if this is a bug or a feature ;).

**Comment by stoicflame** [comment]
Probably a bug. I'll look into it for 2.2.

There are a few things I don't understand yet, though. When you report that the wadl spec says it needs to be a relative API, I think you just mean that it needs to _not_ have a slash at the front. Is that correct?

Then you report that the link to the resource in the generated documentation is incorrect because the browser resolves the absolute path relative to the root and not the context path. But I don't see the connection between the path in the WADL and the path in the generated documentation. Are you saying that it's just that they're both absolute?

Are you providing any special configuration of the application path (e.g. class annotated with `@ApplicationPath`)?

**Comment by vguna** [comment]
Yes, I mean without a slash.

Exactly, both absolute (starting with a slash). So I guess they're using the same "source" for writing it into the WADL as well as for the documentation.

Yes, I'm using:

`@ApplicationPath("/api/1")`

So that seems to be the "source" :).

But that seems ok, I guess?

http://stackoverflow.com/questions/14179868/resteasy-not-working-with-applicationpath

On google I can find both types. With and without a starting slash.

So maybe enunciate can simply drop it?

**Comment by stoicflame** [comment]
Fixed at 944969b.

**Comment by stoicflame** [comment]
[Enunciate 2.2 is now available](https://github.com/stoicflame/enunciate/releases/tag/v2.2.0).

**Comment by hlavki** [comment]
I think it doesn't work, because I have @ApplicationPath("api") and resource with @Path("library") and wadl is generated without slash:
`<wadl:resource path="apilibrary/books">`

**Comment by stoicflame** [comment]
Okay, tracking at #316 .

**Comment by ke5stl** [comment]
agreed.  I have a case where the @ApplicationPath("api/v1") and @Path("customer") appear in the wadl as:

path="api/v1customer" which is not correct.  seems to be removing the leading slash from all parts, while it should perhaps remove only from the first?

**Comment by vguna** [comment]
As per wadl spec http://www.w3.org/Submission/wadl/#x3-120002.6 the path must be relative. Or I'm missing something here?

**Comment by stoicflame** [comment]
Fixed again at 3477047.
